service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/ConsentMapper.java

```diff
@@ -0,0 +1,190 @@
+package uk.nhs.adaptors.oneoneone.cda.report.mapper;
+
+import lombok.AllArgsConstructor;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.Consent;
+import org.hl7.fhir.dstu3.model.Consent.ConsentDataComponent;
+import org.hl7.fhir.dstu3.model.Consent.ConsentDataMeaning;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.Identifier;
+import org.hl7.fhir.dstu3.model.Narrative;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Period;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.springframework.stereotype.Component;
+import uk.nhs.adaptors.oneoneone.cda.report.util.DateUtil;
+import uk.nhs.connect.iucds.cda.ucr.CE;
+import uk.nhs.connect.iucds.cda.ucr.IVLTS;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Authorization;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01ClinicalDocument1;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component2;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component5;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Consent;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Entry;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Observation;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Section;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01StructuredBody;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static org.hl7.fhir.dstu3.model.IdType.newRandomUuid;
+
+@Component
+@AllArgsConstructor
+public class ConsentMapper {
+    private static final String OPT_OUT_URI = "http://hl7.org/fhir/ConsentPolicy/opt-out";
+    private static final String FHIR_SNOMED = "http://snomed.info/sct";
+    private static final String ITK_SNOMED = "2.16.840.1.113883.2.1.3.2.4.15";
+    private static final String NPFIT_CDA_CONTENT = "2.16.840.1.113883.2.1.3.2.4.18.16";
+    private static final String SYSTEM_CODE = "887031000000108";
+    private static final String PERMISSION_TO_VIEW = "COCD_TP146050GB01#PermissionToView";
+
+    public Consent mapConsent(POCDMT000002UK01ClinicalDocument1 clinicalDocument, Encounter encounter) {
+        Consent consent = new Consent();
+        consent.setIdElement(newRandomUuid());
+
+        Patient patient = (Patient) encounter.getSubjectTarget();
+
+        Identifier docIdentifier = new Identifier();
+        docIdentifier.setUse(Identifier.IdentifierUse.USUAL);
+        docIdentifier.setValue(clinicalDocument.getSetId().getRoot());
```

Comment:
Please remember to check if the fields are set. SetId is not mandatory, so we need to check clinicalDocument.isSetSetIs()

Reply:
Updated to check the flag before updating.